src/libaktualizr/uptane/imagesrepository.cc

```diff
@@ -65,33 +65,28 @@ bool ImagesRepository::verifySnapshot(const std::string& snapshot_raw) {
 
 bool ImagesRepository::verifyRoleHashes(const std::string& role_data, const Uptane::Role& role) const {
   const std::string canonical = Utils::jsonToCanonicalStr(Utils::parseJSON(role_data));
-  bool hash_exists = false;
+  // Hashes are not required. If present, however, we may as well check them.
+  // This provides no security benefit, but may help with fault detection.
   for (const auto& it : snapshot.role_hashes(role)) {
     switch (it.type()) {
       case Hash::Type::kSha256:
         if (Hash(Hash::Type::kSha256, boost::algorithm::hex(Crypto::sha256digest(canonical))) != it) {
           LOG_ERROR << "Hash verification for " << role.ToString() << " metadata failed";
           return false;
         }
-        hash_exists = true;
         break;
       case Hash::Type::kSha512:
         if (Hash(Hash::Type::kSha512, boost::algorithm::hex(Crypto::sha512digest(canonical))) != it) {
           LOG_ERROR << "Hash verification for " << role.ToString() << " metadata failed";
           return false;
         }
-        hash_exists = true;
         break;
       default:
```

Comment:
The previous code version returns false in 'default' case while the PR version returns true. Is it desired behavior ?

Reply:
Yes, hence the [comment above](https://github.com/advancedtelematic/aktualizr/pull/1162/files#diff-55cff2ab63b7a704e8b46b05a052f4b6R69). The goal of this PR is to make this very hash check non-required, hence it returns true by default. I could've removed it entirely, but for now I decided to leave it.